### PR TITLE
New vpc channel sdk and test of APIG service

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
- In order to support APIG service through terraform, the vpc channel sdk needs to be added.
- The APIG contains:
  - Dedicated instance
  - Application
  - Group
  - Environment
    - Environment variables
  - API
    - ACL
    - Request throttling policy
    - Custom Authorize
    - Responses
  - **VPC Channel**
  - Domains
  - ...

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: 
```
NONE
```

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add some methods to vpc channel sdk:
  - Create
  - Get
  - List
  - Update
  - Delete
  - Add Backend
  - List Backend
  - Remove Backend
2. add test for Create, Get, List, Update and Delete methods.
```

## Acceptance Steps Performed
```
go test -v -run Test
=== RUN   TestCreateV2VpcChannel
--- PASS: TestCreateV2VpcChannel (0.00s)
=== RUN   TestGetV2VpcChannel
--- PASS: TestGetV2VpcChannel (0.00s)
=== RUN   TestListV2VpcChannel
--- PASS: TestListV2VpcChannel (0.00s)
=== RUN   TestUpdateV2VpcChannel
--- PASS: TestUpdateV2VpcChannel (0.00s)
=== RUN   TestDeleteV2VpcChannel
--- PASS: TestDeleteV2VpcChannel (0.00s)
PASS
ok      github.com/huaweicloud/golangsdk/openstack/apigw/v2/channels/testing    0.028s
```

### CODE REVIEW

File: openstack/apigw/v2/channels/requests.go [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+	// This parameter is valid only when the VPC channel type is set to 2. The value range is 1–65535.
+	// This parameter is required if the VPC channel type is set to 2.
+	Port int `json:"port,omitempty"`
+}
+
+type MemberInfo struct {
+	// Backend server ID.
+	// This parameter is valid when the member type is instance.
+	// The value can contain 1 to 64 characters, including letters, digits, hyphens (-), and underscores (_).
+	EcsId string `json:"ecs_id,omitempty"`
```
Comment (ShiChangkuo) [inline openstack/apigw/v2/channels/requests.go:46]: suggest to add `xor` tag, eg.
```
EcsId string `json:"ecs_id,omitempty" xor:"Host"`
```

File: openstack/apigw/v2/channels/requests.go [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+	if err != nil {
+		r.Err = err
+		return
+	}
+	_, r.Err = client.Post(rootURL(client, instanceId), reqBody, &r.Body, nil)
+	return
+}
+
+// Create is a method by which to create function that create a APIG vpc channel.
+func Update(client *golangsdk.ServiceClient, instanceId, chanId string, opts ChannelOptsBuilder) (r UpdateResult) {
```
Comment (ShiChangkuo) [inline openstack/apigw/v2/channels/requests.go:128]: the comment is misleading

File: openstack/apigw/v2/channels/requests.go [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+		return
+	}
+	_, r.Err = client.Put(resourceURL(client, instanceId, chanId), reqBody, &r.Body, &golangsdk.RequestOpts{
+		OkCodes: []int{200},
+	})
+	return
+}
+
+// Create is a method by which to create function that create a APIG vpc channel.
+func Get(client *golangsdk.ServiceClient, instanceId, chanId string) (r GetResult) {
```
Comment (ShiChangkuo) [inline openstack/apigw/v2/channels/requests.go:141]: ditto

File: openstack/apigw/v2/channels/requests.go (hunk 2 above)
Comment (Lance52259) [reply openstack/apigw/v2/channels/requests.go]: the comments has been udpated

Comment (Lance52259) [reply openstack/apigw/v2/channels/requests.go]: the comment has been updated

File: openstack/apigw/v2/channels/requests.go (hunk 1 above)
Comment (Lance52259) [reply openstack/apigw/v2/channels/requests.go]: Got it

**Pull request merged**